src/Codeception/Test/Loader.php

```diff
@@ -76,9 +81,34 @@ public function __construct(array $suiteSettings)
 
     public function getTests(): array
     {
+        if ($this->shard) {
+            $this->shard = trim($this->shard);
+            if (!preg_match('~^\d+\/\d+$~', $this->shard)) {
+                throw new ConfigurationException('Shard must be set as --shard=CURRENT/TOTAL where CURRENT and TOTAL are number. For instance: --shard=1/3');
+            }
+
+            [$shard, $totalShards] = explode('/', $this->shard);
+
+            if ($shard < 1) {
+                throw new ConfigurationException("Incorrect shard index. Use 1/{$totalShards} to start the first shard");
+            }
+
+            if ($totalShards < $shard) {
+                throw new ConfigurationException('Total shards are less than current shard');
+            }
+
+            $chunks = $this->splitTestsIntoChunks($totalShards);
+
+            return $chunks[$shard - 1] ?? [];
+        }
         return $this->tests;
     }
 
+    private function splitTestsIntoChunks($chunks)
```

Comment:
```suggestion
    private function splitTestsIntoChunks($chunks): array
```

Reply:
Tests failed with `PHP Fatal error:  Uncaught TypeError: Codeception\Test\Loader::splitTestsIntoChunks(): Argument #1 ($chunks) must be of type array, string given, called in /home/runner/work/Codeception/Codeception/src/Codeception/Test/Loader.php on line 100 and defined in /home/runner/work/Codeception/Codeception/src/Codeception/Test/Loader.php:107`